### ISSUE TITLE
Font weight is permanently bold after latest update

### ISSUE THREAD

**Issue by ahmadawais** [microsoft/vscode#214390]
I use `Operator Mono` first this issue occurred in Insider release, and now with the latest release both versions of VS Code only show bold font weight. `300` is too thin. `400` is too bold. This wasn't the case before. 

`400` weight
![image](https://github.com/microsoft/vscode/assets/960133/506b116e-88ab-47c6-bb75-4831f0810f16)

`300` weight
![image](https://github.com/microsoft/vscode/assets/960133/8f351568-807c-475d-9f82-f5a780d60583)

On the prev Version: 1.89.1

`400` weight looks like this. 
![image](https://github.com/microsoft/vscode/assets/960133/e98ab5dc-be37-482d-b3b0-0f202c0e4f73)

Help fix. Thank you! 

Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 
Version: 1.90.0
Commit: 89de5a8d4d6205e5b11647eb6a74844ca23d2573
Date: 2024-06-04T19:34:48.028Z
Electron: 29.4.0
ElectronBuildId: 9593362
Chromium: 122.0.6261.156
Node.js: 20.9.0
V8: 12.2.281.27-electron.0
OS: Darwin arm64 23.5.0

To fix as of now. I'm using the prev version.

1. Deleted the latest app.
2. Installed the previous version: https://code.visualstudio.com/updates/v1_89
3. Disabled auto updates :(

Now all good! That means we need a fix — pretty please.

![image](https://github.com/microsoft/vscode/assets/960133/9b4ccc7e-0571-40f6-8de9-79bbafee4d76)

**Comment by hoangnamitc** [comment]
issues same and I'm rollback to v1.89 to fixed

**Comment by moyus** [comment]
same issue here!

**Comment by Peterabsolon** [comment]
Same issue here. Only certain fonts, like Operator Mono for me

**Comment by mrolfsen** [comment]
Same for me – Operator Mono in my case too.

**Comment by asharirfan** [comment]
Same for me – font is Operator Mono.

**editor.fontWeight: "400"**
![screenshot-2024-06-06-at-07 04 05PM@2x](https://github.com/microsoft/vscode/assets/8023941/bc00356a-e801-4213-9f76-ddc9f774d60c)

**editor.fontWeight: "300"**
![screenshot-2024-06-06-at-07 04 45PM@2x](https://github.com/microsoft/vscode/assets/8023941/d384de2c-55fd-4cdb-b4fb-2715594a9bf1)

**Comment by debuggerpk** [comment]
Same, Operator Mono in my case too.

**Comment by matiasdelagente** [comment]
Same issue, Operator Mono here.

**Comment by leademeule** [comment]
Same issue here with PP Neue Montreal Mono. The 400 font weight now goes to the thin style instead of the book style. Anything above goes to the medium style. This makes everything too thin or too bold.

**Comment by gabemeola** [comment]
Downgrading to April 2024 in the meantime: https://code.visualstudio.com/updates/v1_89

**Comment by ahmadawais** [comment]
@gabemeola how did you downgrade? Install the old one on top? Or is there a better way.

**Comment by ZackSnyder-at** [comment]
Same issue, Operator Mono here.

**Comment by hd4ng** [comment]
@ahmadawais You can use this link to install the previous version: https://code.visualstudio.com/updates/v1_89

**Comment by ahmadawais** [comment]
I did. Still opens the latest one for me. 😅

**Comment by hd4ng** [comment]
@ahmadawais Did you remove the current vscode in the application?

**Comment by ahmadawais** [comment]
I did. Remove as in I moved it out of Applications folder. Think I should delete it then?

**Comment by treyspectre** [comment]
It seems fine here. I only have Operator Mono Book variant.

![Screenshot 2024-06-07 092725](https://github.com/microsoft/vscode/assets/172003642/6264d462-70df-4ab0-8e5a-dad1474f819c)

![Screenshot 2024-06-07 093708](https://github.com/microsoft/vscode/assets/172003642/9b7121ab-8be4-4e89-8ccf-9de18826a5ef)

**Comment by ahmadawais** [comment]
Deleted the old latest app.
Installed the previous version: https://code.visualstudio.com/updates/v1_89

Now all good! That means we need a fix — pretty please. 

![image](https://github.com/microsoft/vscode/assets/960133/63fc624d-18fc-4bd3-b0a3-75af95428928)

**Comment by ZHDreamer** [comment]
> It seems fine here. I only have Operator Mono Book variant.
> 
> ![Screenshot 2024-06-07 092725](https://private-user-images.githubusercontent.com/172003642/337518860-6264d462-70df-4ab0-8e5a-dad1474f819c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc3MzQwMjksIm5iZiI6MTcxNzczMzcyOSwicGF0aCI6Ii8xNzIwMDM2NDIvMzM3NTE4ODYwLTYyNjRkNDYyLTcwZGYtNGFiMC04ZTVhLWRhZDE0NzRmODE5Yy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYwN1QwNDE1MjlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kMDgyZDY3NDk0OGE0NDZlZDJjMTU4OTYwNTJjZGJlMWIwYTg5NjJjYzA1YjM5NjRhYTI2OTcwMzFhZmNiNDUwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.1jx5s3jbmknZ9Zuu36IhJH-65844txJZK_W1EmczNTc)
> 
> ![Screenshot 2024-06-07 093708](https://private-user-images.githubusercontent.com/172003642/337518864-9b7121ab-8be4-4e89-8ccf-9de18826a5ef.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc3MzQwMjksIm5iZiI6MTcxNzczMzcyOSwicGF0aCI6Ii8xNzIwMDM2NDIvMzM3NTE4ODY0LTliNzEyMWFiLThiZTQtNGU4OS04Y2NmLTlkZTE4ODI2YTVlZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYwN1QwNDE1MjlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hZTU3YzA0NTBlMmQzZTZiZDExZjg3ODkxODk2MmI4NmI2ODNkNmYyZWY3ZGYxZDI3YmEyYTlmMWQxNDczNmRiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.GupSJFyKayYAKYHiEBRgIIMpC_5QRvlmk3qKwK-nY5g)

It might be an issue specifically on MacOS

**Comment by deepak1556** [comment]
Sorry for the breakage, issue turned out to be a regression in the runtime which had a migration in font matching api, regressed with [bad commit](https://chromium.googlesource.com/chromium/src/+/4b619e1dae1a1fcf1a41f7364852d6f525abebcc), issue has since then be addressed via [good commit](https://chromium.googlesource.com/chromium/src/+/98eb1392b060711dc30dbc9d1b07fe6c4147e0ff). It is kind of risky to backport the specific commit due to other changes involved, I will instead disable the migrated code path as candidate fix.

Please check if the issue is addressed when starting the application with `--disable-blink-features=FontMatchingCTMigration`

**Comment by Maximization** [comment]
> --disable-blink-features=FontMatchingCTMigration

This flag fixes the issue for me

**Comment by mrwigster** [comment]
> Please check if the issue is addressed when starting the application with `--disable-blink-features=FontMatchingCTMigration`

This flag fixes it for me, thanks. For anyone else wondering, open VS using terminal with the below command:
```
open -a "Visual Studio Code" --args --disable-blink-features=FontMatchingCTMigration
```

**Comment by pascalpp** [comment]
Shout out to all the Wes Bos fans in this thread

**Comment by om4csaba** [comment]
It looks like editing the `.vscode/argv.json` file with "Preferences: Configure Runtime Arguments" does not work. You need to use the command line argument.

**Comment by ahmadawais** [comment]
> Sorry for the breakage, issue turned out to be a regression in the runtime which had a migration in font matching api, regressed with [bad commit](https://chromium.googlesource.com/chromium/src/+/4b619e1dae1a1fcf1a41f7364852d6f525abebcc), issue has since then be addressed via [good commit](https://chromium.googlesource.com/chromium/src/+/98eb1392b060711dc30dbc9d1b07fe6c4147e0ff). It is kind of risky to backport the specific commit due to other changes involved, I will instead disable the migrated code path as candidate fix.
> 
> Please check if the issue is addressed when starting the application with `--disable-blink-features=FontMatchingCTMigration`

@deepak1556 great thanks, will try this out. 

Will this fix be shipped in prod soon? So I can enable to the updates again?

**Comment by deepak1556** [comment]
/fixedWith https://github.com/microsoft/vscode/commit/7640672f4ff8a24c7c5444c0c0ae8ce3ce082b7f

Changes will be available in `1.90.1` release this week, closing for verification.

**Comment by NayamAmarshe** [comment]
This is still not fixed. The fontweights 350 and 400 are exactly the same!

<img width="550" alt="image" src="https://github.com/microsoft/vscode/assets/25067102/549fe234-18e4-40d2-aa23-3f905b16941c">

<img width="553" alt="image" src="https://github.com/microsoft/vscode/assets/25067102/8b44bbd7-d39f-4862-abfd-8ae240d496b3">

**Comment by leademeule** [comment]
@NayamAmarshe on my side font weights seem to now be working correctly. I am also using font weight to select the specific variant of the font I am using. Hopefully this followup comment ([see the previous version screenshots here](https://github.com/microsoft/vscode/issues/214390#issuecomment-2153055011)) can help triangulate the issue?

### Release 1.90.1 (611f9bfce64f25108829dd295f54a6894e87339d): correct font weight applied

<img width="947" alt="1 90 1" src="https://github.com/microsoft/vscode/assets/143581484/7e4565ec-0bf2-4a10-8808-20264eac0939">

**Comment by kylefox** [comment]
Can confirm, seems to be fixed in Version: 1.90.1 for me. Thanks!
